src/RA_Defs.h

```diff
@@ -225,30 +255,382 @@ typedef DWORD			ARGB;
+inline namespace int_literals
+{
+
+// Don't feel like casting non-stop
+// There's also standard literals for strings and clock types
+
+// N.B.: The parameter can only be either "unsigned long long", char16_t, or char32_t
+
+/// <summary>
+///   <para>
+///     Use it if you need an unsigned integer without the need for explicit
+///     narrowing conversions.
+///   </para>
+///   <para>
+///     usage: <c>auto some_int{21_z};</c> where <c>21</c> is
+///            <paramref name="n" />.
+///   </para>
+/// </summary>
+/// <param name="n">The integer.</param>
+/// <returns>The integer as <c>std::size_t</c>.</returns>
+_CONSTANT_VAR operator""_z(unsigned long long n) noexcept {
+    return static_cast<std::size_t>(n);
+} // end operator""_z
+
+
+/// <summary>
+///   <para>
+///     Use it if you need an integer without the need for explicit narrowing
+///     conversions.
+///   </para>
+///   <para>
+///     usage: <c>auto some_int{1_i};</c> where <c>1</c> is
+///            <paramref name="n" />.
+///   </para>
+/// </summary>
+/// <param name="n">The integer.</param>
+/// <returns>The integer as <c>std::intptr_t</c>.</returns>
+_CONSTANT_VAR operator""_i(unsigned long long n) noexcept {
+    return static_cast<std::intptr_t>(n);
+} // end operator""_i
+
+  // We need one for DWORD, because it doesn't match LPDWORD for some stuff
+_CONSTANT_VAR operator""_dw(unsigned long long n) noexcept {
+    return static_cast<_CSTD DWORD>(n);
+} // end operator""_dw
+
+// streamsize varies as well
+/// <summary>
+/// A literal
+/// </summary>
+/// <param name="n">The n.</param>
+/// <returns></returns>
+_CONSTANT_VAR operator""_ss(unsigned long long n) noexcept {
+    return static_cast<std::streamsize>(n);
+} // end operator""_ss
+
+// Literal for unsigned short
+_CONSTANT_VAR operator""_hu(unsigned long long n) noexcept {
+    return static_cast<std::uint16_t>(n);
+} // end operator""_hu
+
+
+_CONSTANT_VAR operator""_dp(unsigned long long n) noexcept {
+    return static_cast<DataPos>(n);
+} // end operator""_dp
+
+// If you follow the standard every alias is considered mutually exclusive, if not don't worry about it
+// These are here if you follow the standards (starts with ISO C++11/C11) rvalue conversions
+
+
+_CONSTANT_VAR operator""_ba(unsigned long long n) noexcept
+{
+    return static_cast<ByteAddress>(n);
+} // end operator""_ba
+
+
+_CONSTANT_VAR operator""_achid(unsigned long long n) noexcept {
+    return static_cast<AchievementID>(n);
+} // end operator""_achid
+
+
+_CONSTANT_VAR operator""_lbid(unsigned long long n) noexcept {
+    return static_cast<LeaderboardID>(n);
+} // end operator""_lbid
+
+
+_CONSTANT_VAR operator""_gameid(unsigned long long n) noexcept {
+    return static_cast<GameID>(n);
+} // end operator""_gameid
+
+} // inline namespace int_literals
+
+
+inline namespace conversions {
+
+// Noticed some weird uses of rvalues returning as lvalues, whatever they aren't used anywhere
```

Comment:
this comment should be removed, it's more appropriate as a review comment wherever the problematic code was.

Reply:
sure